cmake/app/boilerplate.cmake

```diff
@@ -602,6 +602,12 @@ include(${ZEPHYR_BASE}/cmake/target_toolchain_flags.cmake)
 set(PROJECT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/zephyr)
 set(PROJECT_SOURCE_DIR ${ZEPHYR_BASE})
 
+# Set TFM_BINARY_DIR. Has to be included before boards are included as some
+# boards need to define TFM build rules on the output files.
+if(CONFIG_BUILD_WITH_TFM)
+  set(TFM_BINARY_DIR ${CMAKE_BINARY_DIR}/tfm)
+endif()
```

Comment:
Could you describe exactly what the problem is ?

Starting to place such specific knowledge related to a Zephyr module in boilerplate code seems wrong to me.

If someone needs to know the build folder for TF-M, why can they not use:
```
$<TARGET_PROPERTY:tfm,_EP_BINARY_DIR>
```
to obtain the path.

Or TF-M can set the property `TFM_BINARY_DIR` directly on the `tfm` target, which allows users to do:
```
$<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>
```

A custom property has the advantage of being more visible to users and we can easier document it.

Reply:
Thank you. That was a better suggestion. Will update to a custom target property.

Reply:
@tejlmand This created the folder: `build/tfm/generated/$<TARGET_PROPERTY:tfm_cmake,TFM_BINARY_DIR>/dummy_partition/psa_manifest/tfm_dummy_partition.h`
This is coming from tfm_manifest_list.yaml.in.

Reply:
I was able to fix this by using get_target_properties to store it in a variable.

Reply:
> I was able to fix this by using get_target_properties to store it in a variable.

yes, because generator expressions are not supported inside a file used by configure file, like was tried here:
https://github.com/zephyrproject-rtos/zephyr/blob/206fddecd92efe99aff1495354bdbb6161ac08fa/samples/tfm_integration/tfm_secure_partition/dummy_partition/tfm_manifest_list.yaml.in#L18

The solution you use are safe, because you do it after `find_package(Zephyr)`, so the target is known to have been created. Alternative would have been a more complex use of `add_custom_command()` to generate the needed file.

So I like this solution.